mock/src/lib.rs

```diff
@@ -35,18 +32,18 @@ fn load_file_with_code(imports: &str, code: &str, assembly_file: PathBuf) -> Str
 }
 
 /// Inject `code` along side the specified file and run it
-pub fn run_tx<A>(
-    program: Program,
-    stack_inputs: StackInputs,
-    mut adv: A,
-) -> Result<Process<DefaultHost<A>>, ExecutionError>
-where
-    A: AdviceProvider,
-{
-    // mock account method for testing from root context
-    adv.insert_into_map(Word::default(), vec![Felt::new(255)]).unwrap();
+pub fn run_tx(tx: &PreparedTransaction) -> Result<Process<MockHost>, ExecutionError> {
```

Comment:
Why is `run_tx` a wrapper over `run_tx_with_inputs`? Shoudln't we remove run_tx and call `run_tx_with_inputs` directly?

Reply:
In vast majority of cases (like 99%) we don't need to provide extra inputs. So, it would make the code a bit more confusing if we called this function passing `None` as the second parameter almost all the time.